src/components/cards/CreatePool/ChooseWeights.vue

```diff
@@ -0,0 +1,213 @@
+  upToLargeBreakpoint.value ? 56 : 64
+);
+
+onMounted(async () => {
+  // retrieving height causes reflow, get the height of the wrapper once
+  // and manually uptick it when we add items to prevent double reflow during anim
+  wrapperHeight.value = tokenWeightListWrapper.value?.offsetHeight || 0;
+
+  // add in the first token list item
+  addTokenToPool();
+  addTokenToPool();
+
+  // wait for vue to reflect the changes of above
+  await nextTick();
+
+  distributeWeights();
+});
+
+const handleWeightChange = (weight: string, id: number) => {
+  const tokenWeight = tokenWeights.value[id];
+  tokenWeight.weight = Number(weight);
+
+  distributeWeights();
+};
+
+const handleAddressChange = (address: string, id: number) => {
+  const tokenWeight = tokenWeights.value[id];
+  tokenWeight.tokenAddress = address;
+};
+
+const handleLockedWeight = (isLocked: boolean, id: number) => {
+  const tokenWeight = tokenWeights.value[id];
+  tokenWeight.isLocked = isLocked;
+  distributeWeights();
+};
+
+const addTokenToPool = async () => {
+  // animate the height
+  anime({
+    targets: tokenWeightListWrapper.value,
+    height: `${wrapperHeight.value + tokenWeightItemHeight.value}px`
+  });
+
+  wrapperHeight.value += tokenWeightItemHeight.value;
+
+  const newWeights = [
+    ...tokenWeights.value,
+    { ...emptyTokenWeight, id: tokenWeights.value.length - 1 }
+  ];
+  updateTokenWeights(newWeights);
+
+  // to avoid reflow we are going to transform the totals + add token
+  // down instead of having the new token weight item shift them
+  anime({
+    targets: [totalsRowElement.value, addTokenRowElement.value],
+    translateY: `${tokenWeightItemHeight.value * tokenWeights.value.length}px`
+  });
+
+  await nextTick();
+
+  // get the last added token weight element
+  if (tokenWeightItemElements[tokenWeightItemElements.length - 1]) {
+    anime.set(tokenWeightItemElements[tokenWeightItemElements.length - 1], {
+      left: 0,
+      right: 0,
+      top: `${tokenWeightItemHeight.value *
+        (tokenWeightItemElements.length - 1)}px`,
+      opacity: 0
+    });
+
+    // fade it in as well
+    anime({
+      targets: tokenWeightItemElements[tokenWeightItemElements.length - 1],
+      opacity: 1
+    });
+  }
+
+  distributeWeights();
+};
+
+const distributeWeights = () => {
+  // get all the locked weights and sum those bad boys
+  const lockedPct = sum(
+    tokenWeights.value.filter(w => w.isLocked).map(w => w.weight / 100)
+  );
+  const pctAvailableToDistribute = 1 - lockedPct;
+  const unlockedWeights = tokenWeights.value.filter(w => !w.isLocked);
+  const evenDistributionWeight =
+    pctAvailableToDistribute / unlockedWeights.length;
+  for (const tokenWeight of unlockedWeights) {
+    tokenWeight.weight = Number((evenDistributionWeight * 100).toFixed(2));
+  }
+};
+
+const totalWeight = computed(() => {
+  return Math.floor(
+    sumBy(
+      tokenWeights.value.filter(w => w.tokenAddress !== ''),
+      w => w.weight
+    )
+  );
+});
```

Comment:
To keep all components similar and easy to parse lets group different types of properties together and structure them in a common way. My preference is

1. State
2. Composables
3. Computed
4. Methods
5. Watchers
6. Callbacks

This was a best practice in the original Vue api and just helps parsing components you're not familiar with.